recipes/ceres-solver/all/conanfile.py

```diff
@@ -65,23 +64,25 @@ def _configure_cmake(self):
 
     def config_options(self):
         if self.settings.os == "Windows":
-            self.options.remove("fPIC")
+            del self.options.fPIC
 
     def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
         if self.settings.build_type == "Debug" and self.options.use_glog:
             raise ConanInvalidConfiguration("Ceres-solver only links against the release version of glog")
         if self.options.use_glog and not self.options.use_gflags: #At this stage we can't check the value of self.options["glog"].with_gflags so we asume it is true because is the default value
             raise ConanInvalidConfiguration("To depend on glog built with gflags (Default behavior) set use_gflags=True, otherwise Ceres may fail to link due to missing gflags symbols.")
```

Comment:
Why does it raise `ConanInvalidConfiguration`? It should be a warning I guess.

Reply:
could this not check the options/deps of glogs for gflags?

Reply:
options of dependencies cannot be checked in `configure` method (not even in `requirements` method).

Reply:
Ahhh I used it in build and package_info, I thought there was more availability. Conan 2.0 feature?

Reply:
Yes, if it is not reported, it might be useful to have an issue in Conan about this. Most valuable thing would be to provide not just the feature request, but an example to show the pain and why it is needed.

Thanks!